### PR TITLE
Rename JavaIncrementalExecution{Integration -> Performance}Test

### PR DESCRIPTION
**Pull request by wolfs**
I called the test `IntegrationTest` instead of `PerformanceTest` when I introduced it 🤦

### CODE REVIEW

File: build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt [hunk 1]
```diff
@@ -303,7 +303,8 @@ fun Project.propertiesForPerformanceDb(): Map<String, String> {
     } else {
         selectStringProperties(
             PropertyNames.dbUrl,
-            PropertyNames.dbUsername
+            PropertyNames.dbUsername,
+            PropertyNames.dbPassword
```
Comment (lptr) [inline build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt:307]: What does this change have to do with the rename?

File: .teamcity/performance-tests-ci.json [hunk 2]
```diff
@@ -1039,4 +1039,4 @@
       }
     } ]
   } ]
-}
+}
```
Comment (lptr) [inline .teamcity/performance-tests-ci.json:1042]: I think we should not remove the end of file newline here. Wonder why `.editorconfig` doesn't ensure that it stays...

File: .teamcity/performance-test-durations.json [hunk 3]
```diff
@@ -1,548 +1,526 @@
 [ {
-  "scenario" : "org.gradle.performance.regression.android.RealLifeAndroidBuildPerformanceTest.clean assembleDebug with clean transforms cache",
+  "scenario" : "org.gradle.performance.regression.android.AndroidIncrementalExecutionPerformanceTest.abi change",
```
Comment (lptr) [inline .teamcity/performance-test-durations.json:2]: I don't understand all the changes in this file. Why did all the durations change (mostly it's a significant increase)? Why did so many things move around?

Comment (wolfs) [reply .teamcity/performance-test-durations.json:2]: You don't need to understand the changes in the file. I re-created the file from the current timings on CI. That is why things moved around. That is the first commit.

File: build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt (hunk 1 above)
Comment (wolfs) [reply build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt:307]: I needed/wanted to update the performance runtimes before I did the rename. So I needed the build to pick up the password I have in my `gradle.properties` file.

File: .teamcity/performance-tests-ci.json (hunk 2 above)
Comment (wolfs) [reply .teamcity/performance-tests-ci.json:1042]: Jackson's pretty printer doesn't add the newline at the end of the file. Let me see if I can fix this in our code.

Comment (wolfs) [reply .teamcity/performance-tests-ci.json]: I added the change so that our generation code adds the newline.

**Review by lptr — APPROVED** [review]
LGTM!

**Comment by wolfs** [comment]
@bot-gradle test this

**Pull request merged**